context/app/static/js/components/publications/PublicationVignette/hooks.ts

```diff
@@ -0,0 +1,47 @@
+import useSWR from 'swr';
+import { useAppContext } from 'js/components/Contexts';
+import { fillUrls } from './utils';
+import { PublicationVignette } from '../types';
+
+// Fetcher function that lets SWR fetch multiple urls at once
+const multiFetcher = (...urls: string[]) => {
+  const f = (url: string) => fetch(url).then((response) => response.json());
+  return Promise.all(urls.map((url) => f(url)));
+};
+
+type PublicationVignetteConfsInput = {
+  uuid: string;
+  vignette: PublicationVignette;
+  vignetteDirName: string;
+};
+
+export function usePublicationVignetteConfs({ uuid, vignetteDirName, vignette }: PublicationVignetteConfsInput) {
+  const { assetsEndpoint, groupsToken } = useAppContext();
+  // Extract file paths from the vignette object to form the urls to fetch for this vignette
+  const urls = vignette.figures.map(
```

Comment:
In this case, figures could be undefined.

Reply:
Good call 👍 I'm operating under the assumption that `figures` would be an empty list if none are provided as opposed to being null/undefined, but it's better to be safe than sorry.

Reply:
All set; added a `?.` gate before the `map` call and a fallback to an empty list.